lang/python/python-yaml/Makefile

```diff
@@ -8,36 +8,62 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=PyYAML
-PKG_VERSION:=3.11
+PKG_VERSION:=3.12
 PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=http://pyyaml.org/download/pyyaml/
-PKG_MD5SUM:=f50e08ef0fe55178479d3a618efe21db
+PKG_MD5SUM:=4c129761b661d181ebf7ff4eb2d79950
+PKG_HASH:=592766c6303207a20efc445587778322d7f73b161bd994f227adaa341ba212ab
 
 PKG_LICENSE:=MIT
 PKG_LICENSE_FILES:=COPYING
 PKG_MAINTAINER:=Nicolas Thill <nico@openwrt.org>
 
 PKG_BUILD_DEPENDS:=python
 
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(BUILD_VARIANT)-$(PKG_NAME)-$(PKG_VERSION)
+PKG_UNPACK=$(HOST_TAR) -C $(PKG_BUILD_DIR) --strip-components=1 -xzf $(DL_DIR)/$(PKG_SOURCE)
+
 include $(INCLUDE_DIR)/package.mk
 $(call include_mk, python-package.mk)
+$(call include_mk, python3-package.mk)
 
-define Package/python-yaml
+define Package/python-yaml/Default
   SUBMENU:=Python
   SECTION:=lang
   CATEGORY:=Languages
-  TITLE:=YAML parser and emitter for Python
   URL:=http://pyyaml.org/wiki/PyYAML
+endef
+
+define Package/python-yaml
+$(call Package/python-yaml/Default)
+  TITLE:=YAML parser and emitter for Python
   DEPENDS:=+python +libyaml
+  VARIANT:=python
 endef
 
-define Build/Compile
+define Package/python3-yaml
+$(call Package/python-yaml/Default)
+  TITLE:=YAML parser and emitter for Python3
+  DEPENDS:=+python3 +libyaml
+  VARIANT:=python3
+endef
+
+define PyBuild/Compile/Default
```

Comment:
If you agree with this change https://github.com/openwrt/packages/pull/4558
you could switch this to use  `PyBuild/Compile` & `Py3Build/Compile` 
not sure why I didn't do these names in the first place

that change would allow mechanisms like
```
define PyBuild/Compile
< do extra pre-stuff >
$(call PyBuild/Compile/Default)
< do extra post-stuff >
```